### ISSUE TITLE
For the US State and city data counts overlap

### ISSUE THREAD

**Issue by raysalem** [CSSEGISandData/COVID-19#501]
regard to the US
Every state is not reporting data, hence not a row 
city and state data, both have reports, but do they complete overlap?

could we have another column for state , so we can clearly separate state and city

**Comment by jmar777** [comment]
I just ran into this as well. A few observations:

1. If you sum up all the more specific locations (e.g., county or city-level rows), you don't reach the same total suggested by the state-level row.
2. The state-level values are only populated for the current day (presently 03/11/2020). All columns for previous days are zeroed out.
3. If you view the file from earlier commits, these state-level rows didn't exist.

All of this seems to suggest that something when sideways while preparing the most recent import?

**Comment by jmar777** [comment]
Looking into it further, it appears that as of 03/10, the daily_reports csv is rolling all city and county level records into a single state-level record.

I.e., as you compare rows for the US [from the 9th](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_daily_reports/03-09-2020.csv) with their updated values [for the 10th](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_daily_reports/03-10-2020.csv), you'll notice that there is no longer an alignment between them.

**Comment by raysalem** [comment]
but with that said the data validity needs to be maintained. ideally new columns for state, and then rename providence to just city. 
the city roll up should always equal that particular state.

would avoid alot of mis-information. prior to examine the data i was just summing across the nation of "US", but that add 50% increase.

**Comment by lukesneeringer** [comment]
I _think_ this is a byproduct of #504, but this issue is more precisely oriented to the specific problem I commented about there.

Reposting what I said since the links may be useful:

> The current CSV output seems to be the worst of all worlds. Instead of picking one system and going with it, the data seems to be using multiple systems semi-arbitrarily.
> 
> The United States data seems to offer the best example. Originally, the team was using counties, such as [King County in Washington](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv#L35). That was probably too precise for a lot of use cases (King and Snohomish are probably really one event, for example), so now the change seems to be to use states.
> 
> But the [data for Washington state](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv#L245) in the same file is wrong. It says there are 0 cases since the beginning of time and then 267 cases starting yesterday. That is incorrect. I get the desire to use more useful "buckets" for data, but _without migrating the previous data to the new system_, the output becomes difficult to work with. Now I have to figure out how to dedupe this stuff.
> 
> But wait, it gets worse: Now [cities are popping up](https://github.com/CSSEGISandData/COVID-19/blob/master/csse_covid_19_data/csse_covid_19_time_series/time_series_19-covid-Confirmed.csv#L310) starting today. What do I do with this? Is this in addition to the overall Washington state number? Is it duplicative? What county is Kitsap in, and is it duplicative with that county's number?
> 
> Unfortunately, the CSV output just became effectively useless because it contains duplicate information and incomplete history on many rows, and I can not see any good way for a consumer of this data to write code to properly sort it out.

**Comment by Fazin** [comment]
Apparently US data are not just duplicated on the file if we compare the lines with the dates from each state without the Province the date '3/10/20' sums up to a total of 959 cases and at the provinces/state level, the sum is 711.

**Comment by arik-so** [comment]
Looking at the commit history, it appears that the automated data upload, #0cea9b2179306618bd7917798819ebf6608d67de, was filling the rows with the old names (county, state) up until 3/9, with the 3/10 cells being null or 0. For the new names (state only), the cells were all 0 through 3/9, and the new count was present on 3/10.

Now, it appears that the null cell values got replaced with forward propagations from the previous day, or the new data, making dupes really hard to detect programmatically, because I don't know what other sets were renamed. 

A couple other examples of renames I noticed are Vietnam -> Viet Nam, Vatican -> Holy See.

**Comment by bluef00ted** [comment]
@CSSEGISandData: I think others such as @lukesneeringer have explained the problem above but I want to point out the exact error with WASHINGTON state death totals today (3/10) which appears to be everywhere. Cities and states can't co-exist (one or the other has to take over).

Today's **time_series_19-covid-Deaths.csv** file shows the following for 3/10:
QTY Location
21   King County, WA
1   Snohomish County, WA
1   Grant County, WA
0   Clark County, WA
0   Jefferson County, WA
0   Pierce County, WA
0   Kittitas County, WA
23 Washington  <== But with a city like King County, it's not possible to simply sum the columns like before.
0 Kitsap, WA

Is the total deaths for Washington 23 or 46? Nobody can simply 'sum' a column.

Same problem exists in confirmed CSV and it appears is happening everywhere. The CSVs are (sadly) now useless!

Please fix.

**Comment by raysalem** [comment]
To that point, most end users are most interested in large scale date
(countries and maybe states, i recognize people living in those cities
would value there local data). hence there should be a priority put on
national data first. otherwise the data set becomes irrelevant (as
mentioned before)

-Ray

**Comment by bluef00ted** [comment]
> 
> To that point, most end users are most interested in large scale date (countries and maybe states, i recognize people living in those cities would value there local data). hence there should be a priority put on national data first. otherwise the data set becomes irrelevant (as mentioned before)
> […](#)
> -Ray

Agreed! Looking at confirmed CSV, the problem appears much worse than I thought. I think I'm just now coming to grips on how much data is now wrong.

All the states have a confirmed count which you can't obtain by summing up the cities in that state! 

Here's a few examples in the confirmed CSV for today, 3/10/2020:

New York cities: <Sum of all the cities adds to 150>
New York: 173

MA cities: <Sum of all cities add to 41>
Massachusetts: 92

Ouch! I didn't see that. I can't even creatively use the sheets anymore since it appears data is now lost.

**Comment by bluef00ted** [comment]
After further review, the questions which need to be confirmed by CSSE (or whoever knows) seems to be:
 1) FOR_CONFIRMED: Is it OK to sum the entire column in time-series CSV because the # in a state row are just cases with no defined city?
 2) FOR DEATHS: Simply ignore state when doing a column sum?

While it's inconsistent, at least I don't worry the data is wrong. But who knows if this is acceptable?
